internal/appsec/emitter/trace/service_entry_span.go

```diff
@@ -55,19 +55,31 @@ func (op *ServiceEntrySpanOperation) SetTag(key string, value any) {
 	op.tags[key] = value
 }
 
-// SetJSONTag adds the key/value pair to the tags to add to the service entry span. Value will be serialized as JSON.
-func (op *ServiceEntrySpanOperation) SetJSONTag(key string, value any) {
+// SetSerializableTag adds the key/value pair to the tags to add to the service entry span.
+// The value MAY be serialized as JSON if necessary but simple types will not be serialized.
+func (op *ServiceEntrySpanOperation) SetSerializableTag(key string, value any) {
 	op.mu.Lock()
 	defer op.mu.Unlock()
-	op.jsonTags[key] = value
+	switch value.(type) {
+	case string, int8, int16, int32, int64, uint8, uint16, uint32, uint64, float32, float64, bool:
+		op.tags[key] = value
+	default:
+		op.jsonTags[key] = value
+	}
 }
 
-// SetJSONTags adds the key/value pairs to the tags to add to the service entry span. Values will be serialized as JSON.
-func (op *ServiceEntrySpanOperation) SetJSONTags(tags map[string]any) {
+// SetSerializableTags adds the key/value pairs to the tags to add to the service entry span.
+// Values MAY be serialized as JSON if necessary but simple types will not be serialized.
+func (op *ServiceEntrySpanOperation) SetSerializableTags(tags map[string]any) {
 	op.mu.Lock()
 	defer op.mu.Unlock()
-	for k, v := range tags {
-		op.jsonTags[k] = v
+	for key, value := range tags {
+		switch value.(type) {
+		case string, int8, int16, int32, int64, uint8, uint16, uint32, uint64, float32, float64, bool:
+			op.tags[key] = value
+		default:
```

Comment:
TBH this is looking a lot like combining impl. with the singular form would be beneficial IMHO...